examples/miscellaneous/plot_anomaly_bench.py

```diff
@@ -0,0 +1,160 @@
+"""
+==================================
+Anomaly detection's ROC benchmarks
+==================================
+
+This example benchmarks outlier detection algorithms using ROC curves
+on classical anomaly detection datasets. The algorithm performance
+is assessed in an outlier detection context:
+
+1. The algorithms are trained on the whole dataset which is assumed to
+contain outliers.
+
+2. The ROC curve is computed on the same dataset using the knowledge
+of the labels.
+
+
+Interpreting the ROC plot
+-------------------------
+The algorithm performance relates to how good the true positive rate (TPR)
+is at low value of the false positive rate (FPR). The better algorithm
+have the curve on the top-left of the plot and the area under curve (AUC)
+close to 1. The diagonal dashed line represents a random classification
+of outliers and inliers.
+"""
+
+from time import time
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.neighbors import LocalOutlierFactor
+from sklearn.ensemble import IsolationForest
+from sklearn.metrics import roc_curve, auc
+from sklearn.datasets import fetch_kddcup99, fetch_covtype, fetch_openml
+from sklearn.preprocessing import LabelBinarizer
+
+print(__doc__)
+
+random_state = 1  # to control the random selection of anomalies in SA
+
+# datasets
+datasets = ["http", "smtp", "SA", "SF", "forestcover",
+            "glass", "wdbc", "cardiotocography"]
+
+# outlier detection models
+models = [
+    ("LOF", LocalOutlierFactor(n_neighbors=20, contamination="auto")),
+    ("IF", IsolationForest(random_state=random_state,
+                           contamination="auto")),
+]
+
+plt.figure(figsize=(5, len(datasets) * 3))
+for dataset_idx, dataset_name in enumerate(datasets):
+    plt.subplot(len(datasets), 1, dataset_idx + 1)
+
+    # loading and vectorization
+    print("loading data: ", str(dataset_idx + 1))
+    if dataset_name in ["http", "smtp", "SA", "SF"]:
+        dataset = fetch_kddcup99(
+            subset=dataset_name, percent10=True, random_state=random_state
+        )
+        X = dataset.data
+        y = dataset.target
+        if dataset_name == "SA":
+            idx = np.random.choice(
```

Comment:
why is this needed? also if we need it you should use `random_state`

Reply:
Sure I can add `np.random.seed(random_state)`. For your question, `SA` has very high dimensions (~41) comparing to `http`, `smtp`, and `SF`. I picked fewer samples to reduce run time.

Reply:
Solved